aiida/plugins/entry_point.py

```diff
@@ -11,18 +11,31 @@
 import enum
 import traceback
 import functools
-
-try:
-    from reentry.default_manager import PluginManager
-    # I don't use the default manager as it has scan_for_not_found=True
-    # by default, which re-runs scan if no entrypoints are found
-    ENTRYPOINT_MANAGER = PluginManager(scan_for_not_found=False)
-except ImportError:
-    import pkg_resources as ENTRYPOINT_MANAGER
+import importlib
+from importlib.metadata import EntryPoint
 
 from aiida.common.exceptions import MissingEntryPointError, MultipleEntryPointError, LoadingEntryPointError
 
-__all__ = ('load_entry_point', 'load_entry_point_from_string')
+__all__ = ('load_entry_point', 'load_entry_point_from_string', 'EntryPoint')
+
+
+class EPManager:
+    """Entrypoint manager class
+
+    Replaces former class from reentry
+    """
+
+    def __init__(self):
+        self._map = None
+
+    def get_entry_point_map(self):
+        if self._map is None:
+            self._map = importlib.metadata.entry_points()
+
+        return self._map
```

Comment:
since you're handing out r/w permission on `self._map` with it and only need the function for lazy initialization, I'd use:

```suggestion
    @property
    def map(self):
        if self._map is None:
            self._map = importlib.metadata.entry_points()

        return self._map
```